pkg/core/managers/apis/mesh/mesh_manager.go

```diff
@@ -119,10 +117,7 @@ func (m *meshManager) Delete(ctx context.Context, resource core_model.Resource,
 			return err
 		}
 	}
-	// delete all secrets
-	if err := m.otherManagers.DeleteAll(ctx, &system.SecretResourceList{}, core_store.DeleteAllByMesh(opts.Name)); err != nil {
-		return errors.Wrap(err, "could not delete associated secrets")
-	}
+	// secrets are deleted via owner reference
```

Comment:
Did we have a bug on k8s of not cleaning up the secrets?

Reply:
No, we never had a bug but we had 2 different ways of cleaning up the resources. Now we have one.

Reply:
IIUC secret deletion though the meshManager is happening only on Universal? How did we clean up secrets on k8s?

Reply:
> IIUC secret deletion through the meshManager is happening only on Universal?

I did not check that I thought https://github.com/kumahq/kuma/blob/dd95f22e4e568ebeb899c187b107020a28703db5/pkg/core/managers/apis/mesh/mesh_manager.go#L123-L125 is called always and it just picks which store (k8s or postgres) it uses.